File: 29-03-18/c++/02-Roy-and-Ropes.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
    int t;
    cin >> t;
    while(t--)
    {
        int n;
        cin >> n;
        int dp[n+1] = {0};
        
        for(int i = 1; i < n; i++)
        {
            cin >> dp[i];
        }
        
        for(int i = 1; i < n; i++)
        {
            int val;
            cin >> val;
            dp[i] = max(dp[i], val);
        }
        
        int maxi = 0;
        for(int i = n - 1; i >= 1; i--)
        {
            dp[i] = dp[i] - (n-i);
            if(dp[i] > maxi)
                maxi = dp[i];
        }
        cout << n + maxi << endl;
    }
}

File: 29-03-18/python/02-Roy-and-Ropes.py
for _ in range( int(input()) ):
    n = int(input())
    ar = [int(x) for x in input().split()]
    dp = [int(x) for x in input().split()]
    
    for i in range(0, n-1):
        dp[i] = max(dp[i], ar[i])
    
    maxi = 0
    
    for i in range(n-2, -1, -1):
        dp[i] = dp[i] - (n - i - 1)
        if(dp[i] > maxi):
            maxi = dp[i]
    
    print(n + maxi)
